libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx

```diff
@@ -0,0 +1,437 @@
+    const [offset, setOffset] = useState(0);
+    const [limit, setLimit] = useState(LIMIT);
+    const [totalCount, setTotalCount] = useState(LIMIT);
+    const [filterOptions, setFilterOptions] = useState<Array<IAttributeElement>>([]);
+    const [title, setTitle] = useState("");
+    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+    const dropdownRef = useRef<Dropdown>(null);
+
+    useEffect(() => {
+        getElementTotalCount();
+        loadAttributeTitle(false);
+    }, []);
+
+    useEffect(() => {
+        setValidElements(null);
+        setError(null);
+        setIsLoading(false);
+        setOffset(0);
+        setLimit(LIMIT);
+
+        getElements();
+    }, [props.workspace, searchString]);
+
+    const loadAttributeTitle = async (force = false) => {
+        if (!force && isLoading) {
+            return;
+        }
+
+        setIsLoading(true);
+        setError(null);
+
+        try {
+            const attributes = getBackend().workspace(props.workspace).attributes();
+            const displayForm = await attributes.getAttributeDisplayForm(getObjRef());
+            const attribute = await attributes.getAttribute(displayForm.attribute);
+
+            setTitle(attribute.title);
+            setIsLoading(false);
+        } catch (error) {
+            setTitle("");
+            setError(error);
+            setIsLoading(false);
+
+            props.onError(error);
+        }
+    };
+
+    const loadElements = async (offset: number, limit: number) => {
+        const { workspace } = props;
+
+        setIsLoading(true);
+
+        const newElements = await getBackend()
+            .workspace(workspace)
+            .attributes()
+            .elements()
+            .forDisplayForm(getObjRef())
+            .withOptions({
+                ...(searchString ? { filter: searchString } : {}),
+            })
+            .withOffset(offset)
+            .withLimit(limit)
+            .query();
+
+        const mergedValidElements = mergeElementQueryResults(validElements, newElements);
+        const { items } = mergedValidElements;
+
+        // make sure that selected items have both title and uri, otherwise selection in InvertableList won't work
+        // TODO we could maybe use the InvertableList's getItemKey and just use title or uri for example
+        const updatedSelectedItems = updateSelectedOptionsWithData(selectedFilterOptions, items);
+        const updatedPrevSelectedItems = updateSelectedOptionsWithData(prevSelectedFilterOptions, items);
+
+        setSelectedFilterOptions(updatedSelectedItems);
+        setPrevSelectedFilterOptions(updatedPrevSelectedItems);
+        setIsLoading(false);
+        setValidElements(searchString ? newElements : mergedValidElements);
+        setTotalCount(firstLoad ? items.length : totalCount);
+        setFirstLoad(false);
+        setFilterOptions(items as Array<IAttributeElement>);
+    };
+
+    const updateSelectedOptionsWithData = (
```

Comment:
This could be made an util: it does not use any props/state from this component, right?

Reply:
function was made an util